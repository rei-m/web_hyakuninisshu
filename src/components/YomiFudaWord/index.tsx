import * as React from 'react';
import shouldUpdate from 'recompose/shouldUpdate';
import styled, { keyframes } from '@src/styles/styled-components';

export const SPACE = 'ã€€';

export interface Props {
  word: string;
  dulation: number;
  onAnimationEnd: () => void;
}

const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  50% {
    opacity: 0.5;
  }
  100% {
    opacity: 1;
  }
`;

const Word = styled.span<{ dulation: number }>`
  animation: ${fadeIn} ${({ dulation }) => dulation}s linear 1;
`;

const YomiFudaWord: React.FC<Props> = ({ word, dulation, onAnimationEnd }) =>
  word === SPACE ? (
    <span>{word}</span>
  ) : (
    <Word dulation={dulation} onAnimationEnd={onAnimationEnd}>
      {word}
    </Word>
  );

export default shouldUpdate((props: Props, nextProps: Props) => props.word !== nextProps.word)(YomiFudaWord);
