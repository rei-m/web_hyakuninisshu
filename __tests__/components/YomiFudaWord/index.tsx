import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import YomiFudaWord, { Props } from '@src/components/YomiFudaWord';

describe('<YomiFudaWord />', () => {
  it('should render component', () => {
    const props: Props = {
      word: '„ÅÇ',
      onAnimationEnd: jest.fn(),
    };
    const renderer = ReactTestRenderer.create(<YomiFudaWord {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
