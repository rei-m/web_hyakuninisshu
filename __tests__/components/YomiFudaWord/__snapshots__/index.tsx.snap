// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YomiFudaWord /> should render component 1`] = `
<span
  className="sc-bdVaJa siDlf"
  onAnimationEnd={[MockFunction]}
>
  „ÅÇ
</span>
`;
